File: app/src/main/java/me/liam/support/ISupportActivity.java
package me.liam.support;

import android.os.Bundle;

import me.liam.anim.FragmentAnimation;

public interface ISupportActivity {

    public void setDefaultAnimation(FragmentAnimation animation);

    public FragmentAnimation getDefaultAnimation();

    public <T extends SupportFragment> T findFragmentByClass(Class cls);

    public void postDataToFragments(int code, Bundle data);

    public void loadRootFragment(int containerId, SupportFragment to, FragmentAnimation anim, boolean playEnterAnim, boolean addToBackStack);

    public void loadRootFragment(int containerId, SupportFragment to);

    public void loadMultipleRootFragments(int containerId, int showPosition, SupportFragment... fragments);

    public void showHideAllFragment(SupportFragment show);

    public void start(SupportFragment to);

    public void start(SupportFragment to, boolean addToBackStack);

    public void start(SupportFragment from, SupportFragment to);

    public void start(SupportFragment from, SupportFragment to,boolean addToBackStack);

    public void startWithPop(SupportFragment from, SupportFragment to);

    public void pop();

    public void popTo(Class cls);

    public void popTo(Class cls, boolean includeTarget);

}

File: core/src/main/java/me/liam/fragmentation/support/ISupportActivity.kt
package me.liam.fragmentation.support

import android.os.Bundle
import androidx.annotation.IdRes

import me.liam.fragmentation.anim.FragmentAnimation

interface ISupportActivity {

    val defaultBackground: Int

    var defaultAnimation: FragmentAnimation

    fun <T : SupportFragment> findFragment(cls: Class<T>): T?

    fun postDataToFragments(code: Int, data: Bundle)

    fun loadRootFragment(@IdRes containerId: Int, to: SupportFragment, anim: FragmentAnimation, playEnterAnim: Boolean, addToBackStack: Boolean)

    fun loadRootFragment(@IdRes containerId: Int, to: SupportFragment)

    fun loadMultipleRootFragments(@IdRes containerId: Int, showPosition: Int, vararg fragments: SupportFragment)

    fun showHideAllFragment(show: SupportFragment)

    fun start(to: SupportFragment, addToBackStack: Boolean = true)

    fun start(from: SupportFragment, to: SupportFragment, addToBackStack: Boolean = true)

    fun startWithPop(to: SupportFragment)

    fun startWithPop(from: SupportFragment, to: SupportFragment)

    fun startWithPopTo(to: SupportFragment, cls: Class<*>, includeTarget: Boolean = true)

    fun startWithPopTo(from: SupportFragment, to: SupportFragment, cls: Class<*>, includeTarget: Boolean = true)

    fun pop()

    fun popTo(cls: Class<*>, includeTarget: Boolean = true)

}
